File: output/ChatGPT/codenet/C++/Python/s466967233.py
# Python Code
N, K = map(int, input().split())
A = [-1] + list(map(int, input().split()))
B = [-1] * 200100
now = 1
cnt = 0

B[1] = 0
for j in range(1, K+1):
    now = A[now]
    if B[now] == -1:
        B[now] = j
    else:
        cnt = j
        break

if j < K:
    tmp = (K - B[now]) % (cnt - B[now])
    for j in range(tmp):
        now = A[now]

print(now)

File: output/ChatGPT/codenet/C++/Java/s466967233.java
import java.util.Scanner;

public class s466967233 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long N, K;
        long[] A = new long[200100];
        long[] B = new long[200100];
        int now = 1;
        long cnt = 0;
        N = scanner.nextLong();
        K = scanner.nextLong();
        for (int i = 1; i <= N; i++) {
            A[i] = scanner.nextLong();
        }
        for (int i = 1; i <= 200100; i++) {
            B[i] = -1;
        }
        long j;
        B[1] = 0;
        for (j = 1; j <= K; j++) {
            now = (int) A[now];
            if (B[now] == -1) {
                B[now] = j;
            } else {
                cnt = j;
                break;
            }
        }
        if (j < K) {
            long tmp = (K - B[now]) % (cnt - B[now]);
            for (long i = 0; i < tmp; i++) {
                now = (int) A[now];
            }
        }
        System.out.println(now);
    }
}